.github/actions/build-dotnet/write-cache-configuration.ps1

```diff
@@ -0,0 +1,26 @@
+param($IsNuget, $IsNpm, $RestoreKeys, $Hash)
+        
+function Set-Output($Key, $Value) { Write-Output "::set-output name=$Key::$Value" }
+function Invoke-Maybe($block) { try { Invoke-Command -ScriptBlock $block } catch { } }
+
+$paths = @()
+
+if ($IsNuget) { $paths += ,'~/.nuget/packages' }
+
+if ($IsNpm) 
+{
+    (Invoke-Maybe { pnpm store path }), (npm config get cache) | 
+        ? { -not [string]::IsNullOrEmpty($_) } | 
+        % { $paths += $_ }
+}
+
+# Ensure the paths exist.
+$paths | % { New-Item -ItemType Directory -Force $_ } | Out-Null
+
+# Multiple paths must be separaters by "\n", but we can't include newline in the workflow 
+# command so we have to mususe the format function like this.
```

Comment:
```suggestion
# Multiple paths must be separatered by "\n", but we can't include newline in the workflow # command so we have to
misuse the format function like this.
```

Reply:
So, .ps1 files should be wrapped to 120 columns too?

Reply:
Yes. Unless specified otherwise, general software development and C# coding rules apply.